### ISSUE TITLE
Can't import parts any more

### ISSUE THREAD

**Issue by jakebeal** [SynBioDex/SBOLDesigner#246]
3.0 prerelease, trying to import a CDS: when I select the CDS, the part edit dialogue disappears and I'm back to the top-level screen, with nothing having appeared in the part information from the attempted import.

**Comment by jakebeal** [comment]
It wasn't a file, it was importing from the registry.

**Comment by jakebeal** [comment]
I am trying to import BBa_M45215 from the synbiohub iGEM registry into the CDS of the attached SBOL file (renamed to .txt so github will accept it as an attachment)

[red.sbol.txt](https://github.com/SynBioDex/SBOLDesigner/files/1702884/red.sbol.txt)

**Comment by cjmyers** [comment]
Ok, I can reproduce this now.  So, you imported the file rather than directly from the registry.  @michael13162 when import, it asks to me to select what to import from the file, then it does not actually import the information.

**Comment by jakebeal** [comment]
I first imported the file, then tried to fill in the blank part from the registry.

**Comment by cjmyers** [comment]
Hmm, now I'm not sure I'm understanding.  I just looked into your file, and your CDS does indeed not have a sequence, so I think I was wrong when I said it was not imported from the file.  It was imported as a part without a sequence, which is actually the correct behavior.

Now, if you then try to import BBa_M45215 from the registry, it will overwrite the previously assigned part that you loaded from the file, at least that is what happens for me.   This is intended.  There is not currently a way to "merge" information.  If what you are trying to do is to obtain the sequence from BBa_M45215 and attach it to your new part id, then the approach would be download the part from the registry, then edit the part and agree when you save to move it into your local namespace.

Perhaps I'm still missing the steps you took.

**Comment by jakebeal** [comment]
I do want BBA_M45215 to overwrite, and this is not what happens for me.  When I run the current pre-release 3, it simply fails and leaves me with the same blank part.

**Comment by cjmyers** [comment]
Hmm, this one is very strange.  I cannot reproduce.  I wonder if I’m still missing some step that you are taking.  Are you pulling this part from SynBioHub or Built-in Parts?  If you don’t first associate from a file, can you associate from the registry?  

My guess is it might be related to the namespace issues you have been having, since we don’t typically change them on the fly.   There does appear to be an issue that if you change the domain in the preferences, that I’ve had to restart to get it to register properly.  Maybe that is related, but I’m not sure.

**Comment by michael13162** [comment]
If Chris is right about this being related to the namespace issue, it may be fixed now since https://github.com/SynBioDex/SBOLDesigner/issues/244 is potentially fixed.

**Comment by jakebeal** [comment]
This now imports the part correctly, but then automatically closes not just the import window but also the editing window, which is jarring and a violation of UI principles. Then I have to reopen the part editing window in order to finish my editing.

**Comment by cjmyers** [comment]
Agreed.  @michael13162 is this a small change or a large change?  If small, please execute it before the release of 3.0 otherwise you can delay to 3.1.

**Comment by cjmyers** [comment]
No, we need to keep the buttons here.  Can you just leave the edit window open?  What exactly makes this difficult?

**Comment by michael13162** [comment]
Currently, the only way to have a CD in the PartEditDialog persist in the SBOLDocument is if "save" is clicked.  However, since a part imported from SynBioHub is not in the user's namespace, saving is not allowed.  Therefore, it might make sense to not initially give the user the ability to edit until it's completely persisted in the SBOLDocument?
